### PR TITLE
[Initial solver] add initial heuristic for solving SDVRP via CPLEX

### PR DESCRIPTION
**Pull request by andrgolubev**
* Add initial heuristic that focuses on solving the problem with CPLEX as a Site-Dependent VRP
* Time windows are "considered" when constructing actual routes. The route construction is implemented with simple, straightforward Insertions heuristic

Also changed:
* Extended `Problem` class with constructing allowable vehicles in a more complexity-friendly manner (so that check for allowability is O(1) for known customer and vehicle)
* Added `NDEBUG` flag for release/debug-only code (for example, useful for debug logging)
* Updated initial heuristic interface
* Update `Solution` class to use `std::list` instead of `std::vector` for routes
* Added comparison operators to `TransportationQuantity` class

---
Bottom line:

CPLEX is enabled only for Linux users (the version used is `12.7.1`)

P.S.: you can get CPLEX binaries from me, I have versions for all 3 OSes we use

### CODE REVIEW

**Comment by andrgolubev** [comment]
@seankalejs please, check that the code you submit actually compiles and is correct. I had to fix stuff that was supposed to be updated in [your PR](https://github.com/hardnp/vehicle_routing_problem/pull/17)!

**Comment by Joeyuumy1995** [comment]
hello，what do you mean that CPLEX is enabled only for Linux users? my os is windows...

**Comment by andrey-golubev** [comment]
@Joeyuumy1995 it means that the IBM CPLEX linkage is supported only on Linux-based OS (as of current CMakeLists.txt)
